docs/database_example.ipynb

```diff
@@ -234,26 +265,33 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 8,
+   "execution_count": 9,
    "id": "0b618749",
    "metadata": {},
    "outputs": [],
    "source": [
+    "\n",
+    "\n",
     "@dataclass\n",
     "class ConstantFolding(RewritePattern):\n",
     "\n",
     "    @op_type_rewrite_pattern\n",
     "    def match_and_rewrite(self, op: Addi, rewriter: PatternRewriter):\n",
-    "        if isinstance(op.lhs.op, Constant) and isinstance(op.rhs.op, Constant):\n",
+    "        if (\n",
+    "            isinstance(op.lhs, OpResult) and isinstance(op.lhs.op, Constant) and \n",
```

Comment:
Something I just realize, can we replace this with `if isinstance(op.lhs.owner, Constant)`?